app/components/navBar/navBar.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Created by gaston on 8/8/16.
+ */
+
+var navInst = undefined;
+
+module.exports = navInst ? navInst : navInst = NavBar();
+
+/**
+ * @returns {NavBar}
+ * @constructor
+ */
+function NavBar() {
+  let self = this;
+  var config;
+
+  Object.assign( self, {
+    addBackAction: addBackAction,
+    addOptionButton: addOptionButton,
+    setTitle: setTitle,
+    hide: hide,
+    show: show,
+    addActionButton: addActions,
+    open: open,
+    reset: resetNavBar
+  } );
+
+  init();
+  return self;
+
+  /**
+   * Init NavBar
+   */
+  function init() {
+
+    let navEle = $( '.jjb-navbar' );
+
+    config = {
+      nav: navEle,
+      navHeader: navEle.find( '.navbar-header' ),
+      actionsButtonsContainer: navEle.find( '.jjb-navbar__action-buttons' ),
+      actionsButtons: [],
+      actionsOptions: null,
+      backButton: {
+        icon: 'glyphicon-arrow-left',
+        action: onBack
+      }
+    };
+
+    hide();
+  }
+
+  /**
+   * Open NavBar
+   * @param options with config for buttons
+   */
+  function open( options ) {
+    resetNavBar();
+    if ( options && options.title ) {
+      setTitle( options.title );
+    }
+    if ( options && options.optionButton ) {
+      addOptionButton( options.optionButton );
+    } else {
+      addOptionButton( config.backButton );
+    }
+    if ( options && options.backAction ) {
+      addBackAction( options.backAction );
+    }
+    if ( options && options.actionButtons ) {
+      addActions( options.actionButtons );
+    }
+    show();
+  }
+
+  /**
+   * Add an action button to the left in the nav.
+   * Only one button can be on the left, if you add more than one, the last will be replace.
+   * @param menuButton
+   */
+  function addOptionButton( menuButton ) {
+    let optionButton;
+    config.navHeader.find( '.jjb-navbar__option' ).remove();
+    optionButton = $(
+      `<button class="jjb-navbar__option">
+        <span class="glyphicon ${menuButton.icon}" aria-hidden="true"></span>
+      </button>` );
+    optionButton.click( menuButton.action );
+    config.navHeader.prepend( optionButton );
+  }
+
+  /**
+   * Add an action button to nav bar
+   * @param actions
+   */
+  function addActions( actions ) {
```

Comment:
addActions should always receives an array, since it is plural. If users need a single action, they should pass to this method an array with one element or you can create another method call addAction() for adding a single action.